diff --git a/node_modules/react-native-vision-camera/android/build.gradle b/node_modules/react-native-vision-camera/android/build.gradle
index 88ea6a5..cb4efc1 100644
--- a/node_modules/react-native-vision-camera/android/build.gradle
+++ b/node_modules/react-native-vision-camera/android/build.gradle
@@ -137,6 +137,16 @@ android {
     targetCompatibility JavaVersion.VERSION_1_8
   }
 
+  kotlinOptions {
+    jvmTarget = "1.8"
+    freeCompilerArgs += [
+      "-opt-in=com.facebook.react.bridge.ReactMarker",
+      "-opt-in=com.facebook.react.uimanager.UIManagerHelper",
+      "-opt-in=com.facebook.react.internal.ReactNativeInternalAPI",
+      "-Xsuppress-version-warnings"
+    ]
+  }
+
   externalNativeBuild {
     cmake {
       path "CMakeLists.txt"
